prow/tide/history/history.go

```diff
@@ -105,9 +225,9 @@ func (h *History) AllRecords() map[string][]*Record {
 
 // recordLog is a space efficient, limited size, append only list.
 type recordLog struct {
-	buff  []*Record
-	head  int
-	limit int
+	Buff  []*Record
+	Head  int
+	Limit int
```

Comment:
They are exported so that they can be serialized by the `json` package.
Why should there be a guarantee that the deserialized struct's len(Buff) and Limit are congruent? That isn't required or assumed anywhere.

Reply:
My point is that we should not serialize the json field because it is:
* pointless - you throw it away whenever it differs the desired value when read
* unsafe - there is no guarantee it accurately reflects the data in Buff (aka I could set len(buf) == a billion and limit = 5, or limit = a billion and len(buf) = 5)

Reply:
The value was used during parsing and it would accurately represent the data assuming a human wasn't manually editing it?
In either case I switched to storing a `map[string][]*Record` instead of a `map[string]*recordLog` so this is no longer applicable.